Core/src/ApiHelperTrait.php

```diff
@@ -264,4 +265,45 @@ private function splitOptionalArgs(array $input, array $extraAllowedKeys = []):
 
         return [$input, $optionalArgs];
     }
+
+    /**
+     * Helper method used to validate optons based on the supplied $optionTypes
+     * $optionTypes can be an array of string keys, a protobuf Message classname, or a
+     * the CallOptions classname. Parameters are split and returned in the order
+     * that the options types are provided.
+     */
+    private function validateOptions(array $options, array|Message|string ...$optionTypes): array
+    {
+        $splitOptions = [];
+        foreach ($optionTypes as $optionType) {
+            if (is_array($optionType)) {
+                $splitOptions[] = $this->pluckArray($optionType, $options);
+            } else {
+                if ($optionType === CallOptions::class) {
+                    $callOptionKeys = array_keys((new CallOptions([]))->toArray());
+                    $splitOptions[] = $this->pluckArray($callOptionKeys, $options);
+                } else {
+                    $messageKeys = array_map(
+                        fn ($method) => lcfirst(substr($method, 3)),
+                        array_filter(
+                            get_class_methods($optionType),
+                            fn ($m) => 0 === strpos($m, 'get')
+                        )
+                    );
+                    $messageOptions = $this->pluckArray($messageKeys, $options);
+                    $splitOptions[] = $optionType instanceof Message
+                        ? $this->serializer->decodeMessage($optionType, $messageOptions)
```

Comment:
So this basically means that if the options receive a Message type, now will be an array without types?

Reply:
It means that if a user supplies a `Message` type, then the untyped array is returned as that message object. But if it returns the _classname_ of the message, then it just returns the untyped array representing that message's properties.

I don't really like that second use-case, so I removed it in the Spanner PR. There are a few cases where it's nice to have, but I decided to remove it.